assets/src/components/conversations/StartConversationButton.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+
+import {Button, Modal, TextArea, Tooltip} from '../common';
+import {SendOutlined} from '../icons';
+import * as API from '../../api';
+import logger from '../../logger';
+import {Conversation} from '../../types';
+import {TooltipPlacement} from 'antd/lib/tooltip';
+
+export type Props = {
+  customerId: string;
+  disabled: boolean;
+  disabledTooltipPlacement: TooltipPlacement;
+  disabledTooltipTitle: string;
+  onInitializeNewConversation?: (conservation: Conversation) => void;
+};
+
+export type State = {
+  isConversationModalOpen: boolean;
+  isSendingMessage: boolean;
+  message: string;
+};
+
+export class StartConversationButton extends React.Component<Props, State> {
+  static defaultProps = {
+    disabled: false,
+    disabledTooltipPlacement: 'right',
+    disabledTooltipTitle: 'This customer already has an open conversation',
+  };
+
+  state: State = {
+    isConversationModalOpen: false,
+    isSendingMessage: false,
+    message: '',
+  };
+
+  handleOpenNewConversationModal = () => {
+    this.setState({isConversationModalOpen: true});
+  };
+
+  handleCloseNewConversationModal = () => {
+    this.setState({isConversationModalOpen: false});
+  };
+
+  initializeNewConversation = async () => {
+    const {customerId, onInitializeNewConversation} = this.props;
+    const {message} = this.state;
+
+    this.setState({isSendingMessage: true});
+
+    try {
+      const conversation = await API.createNewConversation(customerId, {
+        message: {
+          body: message,
+          sent_at: new Date().toISOString(),
+        },
+      });
+
+      if (onInitializeNewConversation) {
+        onInitializeNewConversation(conversation);
+      }
+    } catch (err) {
+      logger.error('Failed to initialize conversation!', err);
+    }
+
+    this.setState({isConversationModalOpen: false, isSendingMessage: false});
+  };
+
+  getButton = () => {
+    const {
+      disabled,
```

Comment:
very minor nitpick - can we call this `isDisabled` instead so it doesn't conflict with the html "disabled" attribute (https://www.w3schools.com/tags/att_disabled.asp)

(i think 99% of the time this doesn't matter, but i remember at stripe we had a couple issues with calling a prop `disabled` and had a rule that we would preferably use `isDisabled` instead)